plugins/server.go

```diff
@@ -0,0 +1,70 @@
+package plugins
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop"
+	"github.com/smartcontractkit/chainlink/v2/core/services"
+)
+
+// StartServer returns a started Server.
+// The caller is responsible for calling Server.Stop().
+func StartServer(loggerName string) *Server {
+	s := Server{
+		// default prometheus.Registerer
+		GRPCOpts: loop.SetupTelemetry(nil),
+	}
+
+	lggr, err := loop.NewLogger()
+	if err != nil {
+		fmt.Printf("Failed to create logger: %s\n", err)
+		os.Exit(1)
+	}
+	lggr = logger.Named(lggr, loggerName)
+	s.Logger = logger.Sugared(lggr)
+
+	envCfg, err := GetEnvConfig()
+	if err != nil {
+		lggr.Fatalf("Failed to get environment configuration: %s\n", err)
+	}
+	s.PromServer = NewPromServer(envCfg.PrometheusPort(), lggr)
+	err = s.PromServer.Start()
+	if err != nil {
+		lggr.Fatalf("Unrecoverable error starting prometheus server: %s", err)
+	}
+
+	s.Checker = services.NewChecker()
+	err = s.Checker.Start()
+	if err != nil {
+		lggr.Fatalf("Failed to start health checker: %v", err)
```

Comment:
Because the purpose of this is to be a helper that contains the boilerplate so callers don't have to duplicate it all.
Also the callers would have to handle them without a logger, or include logic to understand if the logger was created.

Reply:
fair enough